### PR TITLE
Nemo/v2/1315/add ssoc dates

### PR DESCRIPTION
**Pull request by nemodjuric**
Connects to #1315 
caseflow certification 3

### CODE REVIEW

File: app/models/appeal.rb [hunk 1]
```diff
@@ -152,6 +152,12 @@ def ssoc_match?(date)
     ssoc_documents.any? { |doc| doc.received_at.to_date == date.to_date }
   end
 
+  # This will return an array containing hash of ssoc_dates and ssoc_match?es.
+  # I.e. [{"date"=>"01/01/2010", "match"=>"true"}, {"date"=>"02/02/2012", "match"=>"true"}]
+  def ssoc_dates_with_matches
+    ssoc_dates.map { |item| [["date", item], ["match", ssoc_match?(item)]] }.map { |item| Hash[item] }
```
Comment (amprokop) [inline app/models/appeal.rb]: hmmm. is this more complicated than it has to be?

`ssoc_dates.map { |item| {date: item, match: ssoc_match?(item) }`

Can we just do this?

Comment (nemodjuric) [inline app/models/appeal.rb]: Yes, this is better.

File: app/models/appeal.rb [hunk 2]
```diff
@@ -152,6 +152,12 @@ def ssoc_match?(date)
     ssoc_documents.any? { |doc| doc.received_at.to_date == date.to_date }
   end
 
+  # This will return an array containing hash of ssoc_dates and ssoc_match?es.
+  # I.e. [{"date"=>"01/01/2010", "match"=>"true"}, {"date"=>"02/02/2012", "match"=>"true"}]
+  def ssoc_dates_with_matches
```
Comment (amprokop) [inline app/models/appeal.rb]: can you add a unit test for this new method somewhere in `spec/models/certification_spec.rb`?

Comment (nemodjuric) [inline app/models/appeal.rb]: I will add it as part of #1587

**Pull request merged**